tests/influence/test_influences.py

```diff
@@ -142,29 +128,33 @@ def test_influence_linear_model(
         hessian_regularization=hessian_reg,
     )
 
-    direct_influences = compute_influences(
-        TorchTwiceDifferentiable(linear_layer, loss),
-        *train_data,
-        *test_data,
-        progress=True,
-        influence_type=influence_type,
-        inversion_method="direct",
-        hessian_regularization=hessian_reg,
-    ).numpy()
+    train_data_loader = DataLoader(list(zip(*train_data)), batch_size=40, shuffle=True)
+    input_data = DataLoader(list(zip(*train_data)), batch_size=40)
+    test_data_loader = DataLoader(
+        list(zip(*test_data)),
```

Comment:
Doesn't building a list defeat the purpose of using generators?

Reply:
I have replaced this ugly list(zip(...)) thingy with TensorDataset when the data is already type Tensor. In the cases (like in this test) where the input is a np.ndarray I have left the syntax since the conversion (copying) needs to be done anyway.

Reply:
If you create a list from an ndarray, and pass it to the constructor of the data loader, you are potentially copying data twice, depending on how DataLoader is implemented. If it just stores a reference to the argument, you are ok, but if it copies the data into some internal structure, then... double copy.

Also, DataLoader supports iterable datasets for streaming data, so it should be possible to avoid the extra list construction.

Of course none of this matters in a test with a few dozen samples.